packages/mobile/src/transactions/feed/TransactionFeed.tsx

```diff
@@ -131,11 +93,106 @@ function mapStandbyTransactionToFeedTokenTransaction(tx: StandbyTransaction): Fe
   }
 }
 
+function addFetchedTransactionFunction(
+  fetchedTransactions: TokenTransaction[],
+  setFetchedTransactions: any
+) {
+  return (transactions: TokenTransaction[]) => {
+    const currentHashes = new Set(fetchedTransactions.map((tx) => tx.transactionHash))
+
+    const transactionsWithoutDuplicatedHash = fetchedTransactions.concat(
+      transactions.filter((tx) => !currentHashes.has(tx.transactionHash))
+    )
```

Comment:
would it be more efficient to first add the `transactions` to `fetchedTransactions` then deduplicate? i think then we would have less loops but not sure if it achieves exactly the same result?

Reply:
I think it's the same? Neither should have a material performance impact I think